src/Compilers/Core/Portable/Text/LargeTextWriter.cs

```diff
@@ -65,7 +68,7 @@ public override void Write(string value)
                 {
                     EnsureBuffer();
```

Comment:
💭 Seems like we could improve this scenario by returning the non-null buffer from `EnsureBuffer`

```
var buffer = EnsureBuffer();

// use buffer instead of _buffer...
```